core/dbt/contracts/project.py

```diff
@@ -25,12 +26,7 @@ def _deserialize(cls, value: str) -> "SemverString":
         return SemverString(value)
 
 
-# this supports full semver,
-# but also allows for 2 group version numbers, (allows '1.0').
-register_pattern(
-    SemverString,
-    r"^(0|[1-9]\d*)\.(0|[1-9]\d*)(\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)?$",  # noqa
-)
+sem_ver_pattern = r"^(0|[1-9]\d*)\.(0|[1-9]\d*)(\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)?$"
```

Comment:
Can you add back in the comment so we know what the regex should match?

Reply:
Thanks! Comment has been restored.